lib/puppet/parser/functions/compliance_map.rb

```diff
@@ -239,10 +248,15 @@ def self.lookup_global_silent(param)
         # Super-hack because 3.X raises a Puppet::ParseError complaining about
         # the lookup function
         rescue NoMethodError, Puppet::ParseError
+          reference_map = nil
+
           begin
-            reference_map = call_function('hiera_hash',['compliance_map',nil])
+            if Puppet.version.split('.').first.to_i > 3
+              reference_map = call_function('hiera_hash',['compliance_map',nil])
+            else
+            end
```

Comment:
We can probably drop this at this point and just Puppet 4 it.

It was a backwards compatibility shim for SIMP 5 but we can straight up fork if we need that in the future. I'll try to get this patched soon.